pkg/ccl/backupccl/backup_test.go

```diff
@@ -1553,6 +1553,39 @@ INSERT INTO d.t2 VALUES (ARRAY['hello']);
 			sqlDB.Exec(t, `INSERT INTO d6.tb1 VALUES (ARRAY['v1']::d6._typ1)`)
 		}
 	})
+
+	t.Run("incremental", func(t *testing.T) {
+		_, _, sqlDB, _, cleanupFn := BackupRestoreTestSetup(t, singleNode, 0, InitNone)
+		defer cleanupFn()
+		sqlDB.Exec(t, `
+SET experimental_enable_enums = true;
+CREATE DATABASE d;
+CREATE TYPE d.greeting AS ENUM ('hello', 'howdy', 'hi');
+CREATE TABLE d.t (x d.greeting);
+INSERT INTO d.t VALUES ('hello'), ('howdy');
+`)
+		{
+			// Start out with backing up t.
+			sqlDB.Exec(t, `BACKUP TABLE d.t TO 'nodelocal://0/test/'`)
+			// Alter d.greeting.
+			sqlDB.Exec(t, `ALTER TYPE d.greeting RENAME TO newname`)
+			// Now backup on top of the original back containing d.greeting.
+			sqlDB.Exec(t, `BACKUP TABLE d.t TO 'nodelocal://0/test/'`)
+			// We should be able to restore this backup, and see that the type's
+			// name change is present.
+			sqlDB.Exec(t, `CREATE DATABASE d2`)
+			sqlDB.Exec(t, `RESTORE TABLE d.t FROM 'nodelocal://0/test/' WITH into_db = 'd2'`)
+			sqlDB.CheckQueryResults(t, `SELECT 'hello'::d2.newname`, [][]string{{"hello"}})
+
+			// Now explicitly use the `INCREMENTAL` backup statement.
```

Comment:
FWIW I don't see much need for this test -- of type backups -- to bother testing via both incremental syntaxes - I think the part above is sufficient to prove it works.

If we did want to test more cases though, I might say either adding a value to a type, a cross-db type reference or a db or cluster backup instead of single table -- those all seem more likely to have type-backup interactions than the two different incremental syntaxes. That said, I wouldn't say any of it is a requirement to merge -- the test above is already better than what we have now.

Reply:
> FWIW I don't see much need for this test -- of type backups -- to bother testing via both incremental syntaxes - I think the part above is sufficient to prove it works.

Oh ok, I wasn't sure there was a difference.

> I might say either adding a value to a type

I'll be adding more tests here as more of the `ALTER TYPE/DROP TYPE` commands get implemented.

>  cross-db type reference

I've disabled these :)

>  db or cluster backup instead of single table

These will be easier once `DROP TYPE` lands.